src/VisualStudio/Core/Def/Implementation/InheritanceMargin/InheritanceMarginHelpers.cs

```diff
@@ -135,5 +143,221 @@ public static ImmutableArray<MenuItemViewModel> CreateMenuItemViewModelsForMulti
 
             return builder.ToImmutable();
         }
+
+        private static string GetToolTipTemplateForSingleTarget(InheritanceRelationship headerRelationship)
+            => headerRelationship switch
+            {
+                InheritanceRelationship.ImplementedInterface => ServicesVSResources._0_implements_1,
+                InheritanceRelationship.BaseType => ServicesVSResources._0_is_derived_from_1,
+                InheritanceRelationship.DerivedType => ServicesVSResources._0_derives_1,
+                InheritanceRelationship.InheritedInterface => ServicesVSResources._0_is_inherited_from_1,
+                InheritanceRelationship.ImplementingType => ServicesVSResources._0_is_implemented_by_1,
+                InheritanceRelationship.ImplementedMember => ServicesVSResources._0_implements_1,
+                InheritanceRelationship.OverriddenMember => ServicesVSResources._0_overrides_1,
+                InheritanceRelationship.OverridingMember => ServicesVSResources._0_is_overridden_by_1,
+                InheritanceRelationship.ImplementingMember => ServicesVSResources._0_is_implemented_by_1,
```

Comment:
can you move Implementing/Implemented pars next to each other?

Reply:
Moved